packages/babel-preset-expo/src/__tests__/compiler.test.ts

```diff
@@ -35,4 +44,111 @@ it(`supports functions with discarded return values in try/catch blocks that run
   )!;
 
   expect(code).toMatchSnapshot();
+  expect(code).not.toContain('react.memo_cache_sentinel');
+});
+
+it(`supports memoizing`, () => {
+  // Ensuring the transform doesn't throw.
+  const { code } = babel.transformFileSync(path.resolve(__dirname, 'samples/App.tsx'), options)!;
+
+  expect(code).toMatchSnapshot();
+  expect(code).toContain('react.memo_cache_sentinel');
+});
+
+it(`supports disabling memoizing`, () => {
+  // Ensuring the transform doesn't throw.
+  const { code } = babel.transformFileSync(path.resolve(__dirname, 'samples/App.tsx'), {
+    ...options,
+    caller: getCaller({
+      name: 'metro',
+      supportsReactCompiler: false,
+      engine: 'hermes',
+      platform: 'ios',
+      isDev: false,
+    }),
+  })!;
+
+  expect(code).not.toContain('react.memo_cache_sentinel');
+});
+
+it(`compiles to CJS 1`, () => {
+  const code = babel.transformFileSync(path.resolve(__dirname, 'samples/compile-memo.tsx'), {
+    ...options,
+    caller: getCaller({
+      name: 'metro',
+      supportsReactCompiler: true,
+      supportsStaticESM: false,
+      engine: 'hermes',
+      platform: 'ios',
+      isDev: true,
+    }),
+  })!.code!;
+  expect(code).not.toContain('import {');
+  expect(code).toContain('react.memo_cache_sentinel');
+});
+
+it(`compiles CJS module`, () => {
+  const code = babel.transformFileSync(path.resolve(__dirname, 'samples/compile-memo-cjs.js'), {
```

Comment:
This test is failing and it's unclear if it's related to compiler or not.

Reply:
I've reduced the repro to:
```js
function Thing() {
  return <div />;
}
```
Which outputs a snippet that contains an `import` statement (should be removed by Babel):

```js
var _reactJsxDevRuntime = require(\"react/jsx-dev-runtime\");
var _jsxFileName = \"/Users/evanbacon/Documents/GitHub/expo/packages/babel-preset-expo/src/__tests__/samples/compile-memo-cjs.js\";
import { c as _c } from \"babel-preset-expo/react-compiler-runtime.js\";
function Thing() {const $ = _c(1);let t0;if ($[0] === Symbol.for(\"react.memo_cache_sentinel\")) {
        t0 = /*#__PURE__*/_reactJsxDevRuntime.jsxDEV(\"div\", {}, void 0, false, { fileName: _jsxFileName, lineNumber: 2, columnNumber: 10 }, this);$[0] = t0;} else {t0 = $[0];}return t0;}_c2 = Thing;var _c2;
$RefreshReg$(_c2, \"Thing\");
```

This doesn't occur when `sourceType: 'unambiguous',` is unset.